src/engraving/layout/v0/chordlayout.cpp

```diff
@@ -2327,8 +2329,94 @@ void ChordLayout::layoutChords3(const MStyle& style, const std::vector<Chord*>&
                 }
             }
         }
-        note->setDotY(dotPosition);      // also removes invalid dots
+        if (dotPosition == DirectionV::AUTO) {
+            dotPosition = note->voice() & 1 ? DirectionV::DOWN : DirectionV::UP;
+        }
+        note->setDotPosition(dotPosition);
+    }
+    // Now, we can resolve note conflicts as a superchord
+    Chord* chord = nullptr;
+    for (Chord* c : chords) {
+        if (c->dots()) {
+            chord = c;
+            break;
+        }
     }
+    if (chord && !chord->staff()->isTabStaff(chord->tick())) {
```

Comment:
It would be great to move this code into a separate function

Reply:
Done!